From 0eba97ccc374e3451a5f0447d8e9122cee90a7e6 Mon Sep 17 00:00:00 2001
From: Arend van Spriel <arend@broadcom.com>
Date: Thu, 2 Oct 2014 15:57:44 +0200
Subject: [PATCH 1202/1202] brcmfmac: deal with NL80211_CHANWIDTH_20_NOHT
 setting

The driver can get a channel definition with bandwidth set to
NL80211_CHANWIDTH_20_NOHT. This was no handled in brcmfmac which
results in a warning splat. One scenario in which this is observed
was when running hostapd with configuration 'hw_mode=g'.

Cc: Jon Nettleton <jon.nettleton@gmail.com>
Signed-off-by: Arend van Spriel <arend@broadcom.com>
---
 .../net/wireless/brcm80211/brcmfmac/wl_cfg80211.c  |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/brcm80211/brcmfmac/wl_cfg80211.c b/drivers/net/wireless/brcm80211/brcmfmac/wl_cfg80211.c
index 4d0166a..f9d1517 100644
--- a/drivers/net/wireless/brcm80211/brcmfmac/wl_cfg80211.c
+++ b/drivers/net/wireless/brcm80211/brcmfmac/wl_cfg80211.c
@@ -352,6 +352,7 @@ static u16 chandef_to_chanspec(struct brcmu_d11inf *d11inf,
 	ch_inf.chnum = ieee80211_frequency_to_channel(ch->center_freq1);
 	primary_offset = ch->center_freq1 - ch->chan->center_freq;
 	switch (ch->width) {
+	case NL80211_CHAN_WIDTH_20_NOHT:
 	case NL80211_CHAN_WIDTH_20:
 		ch_inf.bw = BRCMU_CHAN_BW_20;
 		WARN_ON(primary_offset != 0);
-- 
1.7.9.5

